core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushAggregationIntoTableScan.java

```diff
@@ -198,11 +201,13 @@ public Result apply(AggregationNode node, Captures captures, Context context)
         return Optional.of(
                 new ProjectNode(
                         context.getIdAllocator().getNextId(),
-                        TableScanNode.newInstance(
+                        new TableScanNode(
                                 context.getIdAllocator().getNextId(),
                                 result.getHandle(),
                                 newScanOutputs.build(),
                                 scanAssignments,
+                                TupleDomain.all(),
```

Comment:
nit: maybe we can propagate some of the enforced constrains from the original `tableScan` to the new one...

Reply:
I am not saying no, but I will consider this out of scope for this PR. Would that work for you?

Reply:
> I am not saying no, but I will consider this out of scope for this PR. Would that work for you?

Sounds good, thanks!